# Ensure we correctly identify a 64-bit build and
# use "amd64" and "sparcv9" for the 64-bit cpu string.
# Several tests need to be adjusted accordingly.
# Changes were developed in-house, are Solaris specific,
# and are not suitable for upstream.

--- ruby-3.3.10/tool/config.guess
+++ ruby-3.3.10/tool/config.guess
@@ -423,26 +423,19 @@ case $UNAME_MACHINE:$UNAME_SYSTEM:$UNAME
 	GUESS=sparc-hal-solaris2$SUN_REL
 	;;
     sun4*:SunOS:5.*:* | tadpole*:SunOS:5.*:*)
+	# Check CFLAGS for the 64-bit compile option to set the arch
+	SUN_ARCH="sparc"
+	echo $CFLAGS | /usr/gnu/bin/grep -q -w '\-m64' && SUN_ARCH="sparcv9"
 	SUN_REL=`echo "$UNAME_RELEASE" | sed -e 's/[^.]*//'`
-	GUESS=sparc-sun-solaris2$SUN_REL
+	GUESS=$SUN_ARCH-sun-solaris2$SUN_REL
 	;;
     i86pc:AuroraUX:5.*:* | i86xen:AuroraUX:5.*:*)
 	GUESS=i386-pc-auroraux$UNAME_RELEASE
 	;;
     i86pc:SunOS:5.*:* | i86xen:SunOS:5.*:*)
-	set_cc_for_build
 	SUN_ARCH=i386
-	# If there is a compiler, see if it is configured for 64-bit objects.
-	# Note that the Sun cc does not turn __LP64__ into 1 like gcc does.
-	# This test works for both compilers.
-	if test "$CC_FOR_BUILD" != no_compiler_found; then
-	    if (echo '#ifdef __amd64'; echo IS_64BIT_ARCH; echo '#endif') | \
-		(CCOPTS="" $CC_FOR_BUILD -m64 -E - 2>/dev/null) | \
-		grep IS_64BIT_ARCH >/dev/null
-	    then
-		SUN_ARCH=x86_64
-	    fi
-	fi
+	# Check CFLAGS for the 64-bit compile option to set the arch
+	echo $CFLAGS | /usr/gnu/bin/grep -q -w '\-m64' && SUN_ARCH="amd64"
 	SUN_REL=`echo "$UNAME_RELEASE" | sed -e 's/[^.]*//'`
 	GUESS=$SUN_ARCH-pc-solaris2$SUN_REL
 	;;
--- ruby-3.3.10/tool/config.sub
+++ ruby-3.3.10/tool/config.sub
@@ -1028,7 +1028,7 @@ case $cpu-$vendor in
 
 	# Here we normalize CPU types irrespective of the vendor
 	amd64-*)
-		cpu=x86_64
+		# don't change cpu
 		;;
 	blackfin-*)
 		cpu=bfin
